applications/StructuralMechanicsApplication/custom_conditions/line_load_condition.cpp

```diff
@@ -0,0 +1,458 @@
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    return Kratos::make_intrusive<LineLoadCondition<TDim>>(NewId, pGeom, pProperties);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+Condition::Pointer LineLoadCondition<TDim>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    return Kratos::make_intrusive<LineLoadCondition<TDim>>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+Condition::Pointer LineLoadCondition<TDim>::Clone (
+    IndexType NewId,
+    NodesArrayType const& ThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_cond = Kratos::make_intrusive<LineLoadCondition<TDim>>(NewId, GetGeometry().Create(ThisNodes), pGetProperties());
+    p_new_cond->SetData(this->GetData());
+    p_new_cond->Set(Flags(*this));
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+
+//******************************* DESTRUCTOR *****************************************
+/***********************************************************************************/
+
+template<std::size_t TDim>
+LineLoadCondition<TDim>::~LineLoadCondition()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::GetValueOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector< array_1d<double, 3>>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    this->CalculateOnIntegrationPoints( rVariable, rOutput, rCurrentProcessInfo );
+
+    KRATOS_CATCH( "" );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void LineLoadCondition<TDim>::CalculateOnIntegrationPoints(
+    const Variable<array_1d<double, 3>>& rVariable,
+    std::vector< array_1d<double, 3>>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY;
+
+    const auto& r_geometry = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geometry.IntegrationPoints();
+    const IntegrationMethod integration_method = IntegrationUtilities::GetIntegrationMethodForExactMassMatrixEvaluation(r_geometry);
+
+    if ( rOutput.size() != r_integration_points.size() )
+        rOutput.resize( r_integration_points.size() );
+
+    if (rVariable == NORMAL) {
```

Comment:
i would argue that this function does not really make sense. 

what does it mean to take the normal to a 3d line??

Reply:
Imagine a beam, any case, this is for post process, let the user/developer chose